amespahdbpythonsuite/amespahdb.py

```diff
@@ -0,0 +1,707 @@
+                pickle.dump(self.__data, f, pickle.HIGHEST_PROTOCOL)
+
+            # Store the dumped database filename.
+            self.__data['filename'] = md5
+
+            # stop timer and calculate elapsed time
+            elapsed = timedelta(seconds=(time.perf_counter() - tstart))
+            print(f'Elapsed time: {elapsed}\n')
+
+    def pointer(self):
+        return self.__data
+
+    def __getkeybyuids(self, key, uids):
+        """
+        Get a dictionary of PAHdb properties retrieved by keyword for provided UIDs.
+
+        Parameters
+        ----------
+        key : str
+            Database keyword.
+        uids : list of integers
+            List of UIDs.
+
+        Returns
+        -------
+        Dictionary of retrieved properties with UIDs as keys.
+
+        """
+        return copy.deepcopy(dict((uid, self.__data['species'][uid][key]) for uid in uids))
+
+    def subdatabase(self, uids):
+        """
+        Create a subset of the database containing the retrieved UIDs.
+
+        Parameters
+        ----------
+        uids : list of integers
+            List of UIDs.
+
+        Returns
+        -------
+        PAHdb subset dictionary retrieved from the provided UIDs, with UIDs as keys.
+
+        """
+        d = {}
+        for key in self.__data:
+            if key == 'species':
+                d[key] = dict((uid, self.__data['species'][uid]) for uid in uids)
+            else:
+                d[key] = self.__data[key]
+        return copy.deepcopy(d)
+
+    def gettransitionsbyuid(self, uids):
+        """
+        Retrieve and return transitions object based on UIDs input.
+        UIDs can be a list, e.g. the output of search method, or an integer.
+        Calls the :meth:`amespahdbpythonsuite.transitions.transitions` class.
+
+        Parameters
+        ----------
+        uids : list of integers
+            List of UIDs.
+
+        Returns
+        -------
+        transitions object
+
+        """
+        from amespahdbpythonsuite.transitions import transitions
+        if type(uids) == int:
+            uids = [uids]
+
+        return transitions(type=self.__data['database'],
+                           version=self.__data['version'],
+                           data=self.__getkeybyuids('transitions', uids),
+                           pahdb=self.__data,
+                           # pahdb=self.subdatabase(uids),
+                           uids=uids,
+                           model={'type': 'zerokelvin_m',
+                                  'temperature': 0.0,
+                                  'description': ''},
+                           units={'abscissa': {'unit': 1,
+                                               'str': 'frequency [wavenumber]'},
+                                  'ordinate': {'unit': 2,
+                                               'str': 'integrated cross-section [km/mol]'}}
+                           )
+
+    def getlaboratorybyuid(self, uids):
```

Comment:
It is probably best to implement a 'mock' laboratory-class so that the Travis build doesn't fail.